superset/db_engine_specs/base.py

```diff
@@ -177,6 +178,35 @@ class BaseEngineSpec:  # pylint: disable=too-many-public-methods
         ),
     }
 
+    @classmethod
+    def get_dbapi_exception_mapping(cls) -> Dict[Type[Exception], Type[Exception]]:
+        """
+        Each engine can implement and converge its own specific exceptions into
+        Superset DBAPI exceptions
+
+        Note: On python 3.9 this method can be changed to a classmethod property
+        without the need of implementing a metaclass type
+
+        :return: A map of driver specific exception to superset custom exceptions
+        """
+        return {}
```

Comment:
@dpgaspar you should be able to build this automatically, since the SQLAlchemy dialect exposts the DB API module, which should have the exceptions at the top level with standard names.

Something like this (untested):

```python
dbapi = cls.get_engine(database).dialect.dbapi()

return {
  getattr(dbapi, name): getattr(superset.exceptions, f"SupersetDBAPI{name}"
  for name in {"ProgrammingError", "DatabaseError", etc}
}
```

Maybe have this as the base method and allow subclasses to define their own?